### ISSUE TITLE
Rock Band 2 guitar selector switch and overdrive

### ISSUE THREAD

**Issue by AARP41298** [RPCS3/rpcs3#12384]
### Quick summary

Doesnt matter wich device class select, it doesnt fits at all

### Details

Hi, i just found the amazing RB2 Deluxe and start to play with an 360 RB1 guitar
The thing is, to activate the overdrive/star power with guitar in vertical position, i have to remap RS Y+ to R1
And the selector switch is in the LT but it has to remap to Right Stick Up and Down for the correct functionality.
my question is
**Does rpcs3 support LT+ and LT-?**
in other words
i think i can solve this changing the maping "rules"
insted of 0-100 LT to 0-100 RS Up

something like 50-100 LT+ to 0-100 RS Up
and 50-0 LT- to 0-100 RS Down
![guitar map](https://user-images.githubusercontent.com/33246631/180593813-9ce564f5-9581-4a8d-9374-7cd2d6355078.png)

**Comment by MSuih** [comment]
Could you clarify a bit about what you exactly want us to change? We had three people trying to decipher your message but couldn't quite figure it out.

**Comment by YuriNator557** [comment]
Basically the Rock Band guitars have this little slider thing that can be used ingame to change how the guitar tone sounds I believe
![image](https://user-images.githubusercontent.com/76178650/180598430-21b30e7c-9c5c-4812-b965-43db4d2fc12f.png)
There isn't really a way of getting this working in RPCS3 since it seems to do something weird on the right stick to make it work?

To activate Overdrive (same thing as Guitar Hero's Star Power) normally the guitar would need to be tilted up or press the Select button, but currently in RPCS3 it will just activate automatically as soon as you get enough Overdrive, I think they are asking if it is possible to fix this?

**Comment by Darkhost1999** [comment]
I see this as duplicate feature request.
Please add the ability to map more than 1 physical button to 1 emulated button. 
Because I need to map Right stick up and down both to right bumper to get my Xbox 360 guitar to work correctly.

**Comment by AARP41298** [comment]
Sorry @MSuih, i hope this time i could give you a better explain of what is exactly is happening

This is the normal behavior as i remember in 360
NOTE: Ignore the right stick, that becase the whammy and it works great
![guitar glossary](https://user-images.githubusercontent.com/33246631/180661946-69fb24aa-3b1f-4644-84cf-ee6026dafce3.png)

This is the behavior using an 360 controller as a Device Class-Guitar-Rock Band
![controller as guitar](https://user-images.githubusercontent.com/33246631/180662077-767a6178-93bd-43c4-8240-a498fb354352.png)

And this is the behavior using the guitar controller as a Device Class-Guitar-Rock Band changing the LT to RS Down
![mapping LT](https://user-images.githubusercontent.com/33246631/180662128-f9289b36-2e6b-464f-a6e6-57242799dd0a.png)
![guitar as guitar](https://user-images.githubusercontent.com/33246631/180662173-2b94cdc6-a1eb-4ec0-b047-5b3f1962b5c4.png)

As you can see, there is an offset in the effects
I think that the cuoted feature of @Darkhost1999 is the same problem
Thats why im asking for a LT+ and LT- support
 or maybe an "Half LT+" and "Half LT-" support? IDK

PD I found a problem with Solo Buttons, do i need to rename this issue an explain it here, or i need to open a new issue?

Thank you so much for the replies

**Comment by AARP41298** [comment]
Hey @Darkhost1999 wich number is that issue?

**Comment by AARP41298** [comment]
Maybe using xoutput...

![guitar xoutput rpcs3](https://user-images.githubusercontent.com/33246631/180664712-75ecab14-f207-4c04-85cd-345444bed38a.gif)

**Comment by Megamouse** [comment]
I don't understand the problem to be honest.
You can't map a trigger to the whole axis.
But why do you want that in the first place?
Why can't you just use the stick normally?
Why can't you just use the other trigger for the lower half of the stick axis?
Why does RSY+ have to be mapped to R1?

**Comment by sanjay900** [comment]
> I don't understand the problem to be honest. You can't map a trigger to the whole axis. But why do you want that in the first place? Why can't you just use the stick normally? Why can't you just use the other trigger for the lower half of the stick axis? Why does RSY+ have to be mapped to R1?

The different consoles handle the pickup selector differently. On the 360, the selector is mapped to the Left Trigger, while on the ps3, its mapped to the full range of the right stick y axis. I guess the problem here is that no way to map the trigger across the axis, making it impossible to map the pickup selector from a xbox 360 guitar to RPCS3.

If it helps, i was helping with some documentation on all these  guitars, as I have been working on emulating them on a pi pico with console support, so there are some really nice notes here now: https://github.com/TheNathannator/PlasticBand/tree/main/Docs/Instruments/5-Fret%20Guitar/Rock%20Band

**Comment by sanjay900** [comment]
Guitar inputs are weird right, I even recently discovered that even whammy differs on all the guitars *even on the same platform*. For example, xb360 guitars use the full right stick x axis for whammy, while PS3 GH just uses the top half of the axis, and then rock band uses the entire axis but if you don't touch whammy it goes back to 0x7F after a second or so, much like how the selector switch works.

**Comment by AARP41298** [comment]
update

im trying to remap the xinput outside rpcs3 with xinputplus, durazno, x360ce, xoutput
in xoutput and x360 the remap works, but it takes 2 slots of xinput
xinput 1: original guitar
xinput 2: remaped guitar

xinputplus doesnt add the extra slot but the remapping options are too limited

i will experiment maping xinput to be used in rpcs like dinput

**Comment by AARP41298** [comment]
I kinda solve it
1 connect all the instruments (all xinputs are now filled)
2 open x360 and map only guitars
3 use the config*
4 enable mapped device
5 in rpcs3, gamepad settings, select SDL handler and use the "virtual" 360 controller
![image](https://github.com/RPCS3/rpcs3/assets/33246631/f09b1c74-7725-4379-bcd4-18b7f74cc82a)
Now the guitars are mapped correctly

Thank you for the SDL handler

NOTE: the RB1 drums doesnt have button map issues you may simply use xinput


*config
![image](https://github.com/RPCS3/rpcs3/assets/33246631/2823ff34-545b-48ae-bc24-cb2d73ecbc0d)
PD Even calibrating the stick in DX Tweak i cant reach the flanger effect, but the solo modifier and overdrive gesture (starpower)  work like charm

![image](https://github.com/RPCS3/rpcs3/assets/33246631/1f0b98bb-4501-4be7-874d-6740a904d58a)

**Comment by TeaParty777** [comment]
Hi there!

I would like to add some info that maybe could be useful for more users.
Thanks to AARP41298 who started this solution.

I will leave my x360ce preset, FULLY WORKING!, all works as intended, the selector switch, the tilt for "_STAR POWER_", solo buttons which previosly needed strumming unless the selector switch was all the way down (like on the NONE setting).

https://www.mediafire.com/file/cjhzv62qd1oo0xy/PRESET_FENDER_RB2_WITH_XBOX_OFFICIAL_WIRELESS_RECEIVER_BY_FATALRACER.txt/file

another info:
Installed ViGEmBus 1.22.0 needed for x360ce to work.
x360ce v4.17.15.0 (2020-11-15)
Windows 11
**Official Xbox 360 Wireless Receiver + Rock Band 2 Fender Stratocaster model: XBGTS2**

https://www.youtube.com/watch?v=GDlJhniENw8

**Comment by KaijuChomps** [comment]
What I'm honestly curious to know is why in RPCS3, the Slider (detected as Left Trigger) only seems to even register for mapping purposes past the 70% point?

Would it be possible within RPCS3 ever to specific a specific range of the Slider to be mapped to RS Up and the other half to RS Down, to avoid both being triggered constantly by the inputs?

I know this is highly niche as an issue, but this does affect everyone using 360 Guitars to play Rock Band on RPCS3 right now, which is slowly increasing in the number of folks trying to figure it out.

**Comment by SArpnt** [comment]
the wii rockband guitar also maps these to the triggers instead of the joystick. it would be nice if i could add proper offsets and multipliers to each axis.

**Comment by Megamouse** [comment]
@AARP41298 try with #12384
See the description for how it works.
Save the pad settings once to get the new values in the config.
Then set these values here like this:
```
    Left Trigger Step Percent: 20
    Right Trigger Step Percent: 0
    Left Trigger Step Offset Percent: 10
    Right Trigger Step Offset Percent: 0
```

**Comment by Megamouse** [comment]
ah damn. I think I misunderstood the problem again xD.

**Comment by Megamouse** [comment]
I guess #15932 will fix it for real

**Comment by AARP41298** [comment]
Thank you @Megamouse, presing ALT to activate LT+ and LT-  is perfect.
I still cant reach flanger effect, the same behaviour of x360ce, but now i can get rid of it, only map 3 things and we are ready to go.
![image](https://github.com/user-attachments/assets/571bcd27-3c58-4ac8-b1ad-cba7f7935e59)

If I remember correctly, a long time ago I needed to keep the whammy in the middle position in order to map the buttons, I see that is no longer necessary, thanks for that.

PD consider #14577

**Comment by Megamouse** [comment]
I don't understand what's the flanger effect and what the game expects.

**Comment by Megamouse** [comment]
It's possible that our implementation rounds to a slightly different value than the game would usually get from the device.

**Comment by sanjay900** [comment]
If it helps, the raw values from a PS3 rock band guitar are listed here:
https://github.com/TheNathannator/PlasticBand/blob/main/Docs/Instruments/5-Fret%20Guitar/Rock%20Band/General%20Notes.md

**Comment by sanjay900** [comment]
@AARP41298 
When you put the switch in the flanger position, does it act like nothing is selected?

**Comment by AARP41298** [comment]
@sanjay900 

https://github.com/user-attachments/assets/43516fe1-5ab7-4781-9255-78bf83009dab

**Comment by Megamouse** [comment]
I've asked you guys to test my build and compare what's logged...

**Comment by sanjay900** [comment]
> I've asked you guys to test my build and compare what's logged...

Sorry, I'm away on holiday at the moment and don't actually have access to a computer, so I was just asking questions out of curiosity to validate my thoughts on what was potentially happening.

**Comment by sanjay900** [comment]
Looking at it though, I do believe you would need to have a specific patch for 360 guitars if you want the flanger to work on those, since the flanger value on 360 guitars certainly doesn't line up correctly in respect to what the game wants. Not sure how best to approach fixing that though
